src/extension.ts

```diff
@@ -109,7 +110,11 @@ function registerCommands(context: ExtensionContext) {
 		projectProvider = new ProjectProvider(context);
 		ProjectProvider.setInstance(projectProvider);
 	}
-
+    if (projectProvider.getContext().globalState.get('workspaceSaveInProgress') &&
+        projectProvider.getContext().globalState.get('selectedProject') !== undefined) {
+        devCommands.addProjectsToTheDashBoard(projectProvider, projectProvider.getContext().globalState.get('selectedProject') as string);
+        helperUtil.clearDataSavedInGlobalState(projectProvider.getContext());
+    }
```

Comment:
Moved the above code to a method and called as suggested

Reply:
Thank you for moving the code to its own method. The second part of the suggestion was to make this call outside of the `registerCommands` method, since it does not belong in this method. The call to the new method `handleWorkspaceSaveInProgress` should be made from the `activate` method.

Reply:
handleWorkspaceSaveInProgress requires projectProvider as a parameter and i see we have code for getting the instance for projectProvider within the registerCommands methods , thats why i left handleWorkspaceSaveInProgress within registerCommands

Reply:
moved the handleWorkspaceSaveInProgress to activate, please review